script.program.homeassistant/addon.py

```diff
@@ -0,0 +1,463 @@
+	log('Browse by domain started')
+	listing=[]
+	isFolder = True
+
+	#Test connection / get version
+	response = getRequest('/config')
+	if response is not None:
+		log('Response from server: ' + str(response))
+	else:
+		log('Response from server is NONE!')
+
+	#Compose folder list based on addon settings
+	for d in range(len(haDomainSettings)):
+		if haDomainSettings[d]:
+			url = build_url({'mode': 'loadfolder', 'type': 'domain', 'domain': haDomainNames[d]})
+			icon = os.path.join(imgIconResourcePath) + '\\' + haDomainNames[d] + '.png'
+			li = xbmcgui.ListItem(__addon__.getLocalizedString(haDomainTranslations[d]))
+			li.setArt({'icon': icon, 'fanart' : os.path.join(imgFanartResourcePath,'fanart.jpg')})
+			listing.append((url, li, isFolder))
+
+	#Add the favourites folder
+	if __addon__.getSetting('useFavourites') == 'true':
+		url = build_url({'mode': 'loadfolder', 'type': 'favourites', 'domain': 'none'})
+		icon = os.path.join(imgIconResourcePath, 'favourites.png')
+		li = xbmcgui.ListItem('Favourites')
+		li.setArt({'icon': icon, 'fanart' : os.path.join(imgFanartResourcePath,'fanart.jpg')})
+		listing.append((url, li, isFolder))
+
+	#Add the version number as folder
+	url = build_url({'mode': 'config'})
+	icon = os.path.join(imgIconResourcePath, 'config.png')
+	li = xbmcgui.ListItem(__addon__.getLocalizedString(30023) + ': ' + response['version'])
+	li.setArt({'icon': icon, 'fanart' : os.path.join(imgFanartResourcePath,'fanart.jpg')})
+	listing.append((url, li, isFolder))
+
+	xbmcplugin.addDirectoryItems(addon_handle, listing, len(listing))
+	xbmcplugin.endOfDirectory(addon_handle)
+	
+def loadFolder(folderType, domain):
+	response = getRequest('/states')
+
+	searchKey = ''
+
+	if folderType == 'domain':
+		searchKey = domain + '.'
+		createFolderList(searchKey, response, domain, folderType)
+
+	if folderType == 'favourites' or folderType == 'widgets':
+		favourites = read_favourites()
+		if len(favourites) > 0:
+			for entity in favourites:
+				searchKey = entity['entity_id']
+				domain = entity['domain']
+				createFolderList(searchKey, response, domain, folderType)
+		else:
+			show_dialog(__addon__.getLocalizedString(30057))
+			browseByDomain()
+	if domain != 'vacuum':
+		xbmcplugin.addSortMethod(addon_handle, xbmcplugin.SORT_METHOD_LABEL)
+	xbmcplugin.endOfDirectory(addon_handle)
+
+
+def createFolderList(searchKey, response, domain, folderType):
+	listing=[]
+	isFolder = False
+
+	vacuumService = ['start', 'stop', 'return_to_base', 'locate']
+	vacuumServiceTranslations = [30040, 30041, 30042, 30043]
+
+	if folderType == 'widgets':
+		markup = '[LIGHT] '
+	else:
+		markup = '[CR][LIGHT]'
+	for entity in range(len(response)):
+		if searchKey in response[entity]['entity_id']:
+			contextMenuItems = []
+			entity_id = response[entity]['entity_id']
+			entity_state =response[entity]['state']
+			label = response[entity]['attributes']['friendly_name']
+			icon = os.path.join(imgIconResourcePath) + '\\' + domain + '.png'
+
```

Comment:
you might want to run reindent (https://pypi.org/project/reindent/) against this file. Python (PEP) recommends 4 spaces instead of tabs for identation level. https://www.python.org/dev/peps/pep-0008/
This makes your code easier to read.